File: s24_apriori.R
# Apriori

# Data Preprocessing [§24 Lect159: "Apriori in R - Step 1"]
# install.packages('arules') [Note: Anaconda doesn't have the `arules` package]
library(arules)
dataset = read.csv('data/s24_Market_Basket_Optimisation.csv', header = FALSE)
dataset = read.transactions('data/s24_Market_Basket_Optimisation.csv', sep = ',', rm.duplicates = TRUE)
summary(dataset)
itemFrequencyPlot(dataset, topN = 10)

# Training Apriori on the dataset [§24 Lect160: "Apriori in R - Step 2"]
rules = apriori(data = dataset, parameter = list(support = 0.004, confidence = 0.2))

# Visualising the results [§24 Lect161: "Apriori in R - Step 3"]
inspect(sort(rules, by = 'lift')[1:10])

File: s24_apriori.py
# Apriori

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Data Preprocessing [§24 Lect162: "Apriori in Python - Step 1"]
dataset = pd.read_csv('data/s24_Market_Basket_Optimisation.csv', header = None)
transactions = []
for i in range(0, 7501):
    transactions.append([str(dataset.values[i,j]) for j in range(0, 20)])

# Training Apriori on the dataset [§24 Lect163: "Apriori in Python - Step 2"]
from apyori import apriori #  [Note: Anaconda doesn't have `apyori`; portable `apyori.py` file is in repo]
rules = apriori(transactions, min_support = 0.003, min_confidence = 0.2, min_lift = 3, min_length = 2)

# Visualising the results [§24 Lect164: "Apriori in Python - Step 3"]
results = list(rules)